repo/migrations/Migration020.go

```diff
@@ -0,0 +1,342 @@
+	ci "gx/ipfs/QmPvyPwuCgJ7pDmrKDxRtsScJgBaM5h4EpRL2qQJsmXf4n/go-libp2p-crypto"
+	"gx/ipfs/QmS73grfbWgWrNztd8Lns9GCG3jjRNDfcPYg2VYQzKDZSt/go-ipfs-ds-help"
+	"gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
+	ds "gx/ipfs/QmaRb5yNXKonhbkpNxNawoydk4N6es6b4fPj19sjEKsh5D/go-datastore"
+	"gx/ipfs/QmdxUuburamoF6zF9qjeQC4WYcWGbWuRmdLacMEsW8ioD8/gogo-protobuf/proto"
+	"gx/ipfs/QmfVj3x4D6Jkq9SEoi5n2NmoUomLwoeiwnYz2KQa15wRw6/base32"
+
+	dhtpb "github.com/OpenBazaar/openbazaar-go/repo/migrations/helpers/Migration020"
+	"github.com/ipfs/go-ipfs/repo"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+)
+
+// Migration020 runs an IPFS migration which migrates the IPNS records in the
+// datastore.
+type Migration020 struct{}
+
+func (Migration020) Up(repoPath string, dbPassword string, testnet bool) error {
+	// IPFS code errors if it detects an invalid version so we'll migrate the version first.
+	if err := writeIPFSVer(repoPath, 7); err != nil {
+		return fmt.Errorf("bumping repover to 21: %s", err.Error())
+	}
+
+	// Open the repo
+	r, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	// Load the config
+	cfg, err := r.Config()
+	if err != nil {
+		return err
+	}
+
+	// Open our OpenBazaar db and grab the identity key
+	var databaseFilePath string
+	if testnet {
+		databaseFilePath = path.Join(repoPath, "datastore", "testnet.db")
+	} else {
+		databaseFilePath = path.Join(repoPath, "datastore", "mainnet.db")
+	}
+	db, err := sql.Open("sqlite3", databaseFilePath)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+	if dbPassword != "" {
+		p := fmt.Sprintf("PRAGMA key = '%s';", dbPassword)
+		_, err := db.Exec(p)
+		if err != nil {
+			return err
+		}
+	}
+	stmt, err := db.Prepare("select value from config where key=?")
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+	var identityKey []byte
+	err = stmt.QueryRow("identityKey").Scan(&identityKey)
+	if err != nil {
+		return err
+	}
+	identity, err := ipfs.IdentityFromKey(identityKey)
+	if err != nil {
+		return err
+	}
+
+	// Set our key in IPFS config
+	cfg.Identity = identity
+
+	// Migrate record(s)
+	ks := r.Keystore()
+	keys, err := ks.List()
+	if err != nil {
+		return err
+	}
+
+	dstore := r.Datastore()
+
+	sk, err := myKey(r)
+	if err != nil {
+		return err
+	}
+
+	err = applyForKey(dstore, sk)
+	if err != nil {
+		return err
+	}
+
+	for _, keyName := range keys {
+		k, err := ks.Get(keyName)
+		if err != nil {
+			return err
+		}
+		err = applyForKey(dstore, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Migrate OpenBazaar repover
+	if err := writeRepoVer(repoPath, 21); err != nil {
+		return fmt.Errorf("bumping repover to 21: %s", err.Error())
+	}
+	return nil
+}
+
+func (Migration020) Down(repoPath string, dbPassword string, testnet bool) error {
+	// We're downgrading from version 7.
+	fsrepo.RepoVersion = 7
+	r, err := fsrepo.Open(repoPath)
+	fsrepo.RepoVersion = 6
```

Comment:
What does setting these values do?

Reply:
They are copied from the ipfs migration. I assume they are needed.